app/lib/katello/repo_discovery.rb

```diff
@@ -194,13 +193,14 @@ def file_crawl(resume_point)
 
     def should_follow?(path)
       #Verify:
-      # * link's path starts with the base url
+      # * link's path includes the base url
       # * link hasn't already been crawled
       # * link ends with '/' so it should be a directory
       # * link doesn't end with '/Packages/', as this increases
       #       processing time and memory usage considerably
-      return path.starts_with?(@uri.path) && !@crawled.include?(path) &&
-           path.ends_with?('/') && !path.ends_with?('/Packages/')
+      #
+      return path.include?(@uri.hostname) && !@crawled.include?(path) &&
```

Comment:
Looks like the `path` is really the `url`. Please change the name, but better: pass it as a `URI` object and then compare `uri.hostname` to `@uri.hostname`. Also note that hostnames are case insensitive and plain string comparisons are wrong.